### PR TITLE
Denis courselets

### PR DESCRIPTION
**Pull request by DenisKolesnik**
SignUp tests through Social Networks and Simple SignUp

### CODE REVIEW

File: envs/local.env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+PROJECT=courselets
```
Comment (rimikt) [inline envs/local.env]: Remove this file from git.

File: envs/local.env.example [hunk 2]
```diff
@@ -1,5 +0,0 @@
-PROJECT=<project_name>
```
Comment (rimikt) [inline envs/local.env.example]: Why did you delete this file?

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from tempmail import TempMail
+import json
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Pep8.

Should be:

```
import json

from tempmail import TempMail
```

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+# generate temporary email
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Move this to doctsring.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+def temporary_email_generate():
+    tm = TempMail()
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Add docstring.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    email = tm.get_email_address()
+    print('Your email is: ', email)
+
+    return email
+
+# check the inbox
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Move this to docstring.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def get_mails(email_address):
+    tm = TempMail()
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Docstring.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    text = dumped.split()
+
+    if text[0] == '{"error":':
+        print('the inbox is empty')
+    else:
+        print('inbox isn\'t empty')
+
+    return inbox
+
+# upload json to python and split string
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Move to docstring.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def get_link(mails):
+    data = json.dumps(mails)
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Same as above.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    # data_string = str(mails)
+    split_email = data.split()
+
+    return split_email
+
+# cut string and extract the link
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: Same as above.

File: projects/courselets/features/utils/temp_mail/email_confirmation.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def get_clear_link(split_email):
+    link = []
```
Comment (rimikt) [inline projects/courselets/features/utils/temp_mail/email_confirmation.py]: -_-

Comment (rimikt) [reply projects/courselets/features/utils/temp_mail/email_confirmation.py]: I can't understand what is going on here.

File: projects/courselets/features/pages/main_page.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+"""
+Class to work with main page
+
+"""
+from .base_page import BasePage
+
+
+class MainPage(BasePage):
+
+    def page_url(self):
```
Comment (rimikt) [inline projects/courselets/features/pages/main_page.py]: This can be a property.

File: projects/courselets/features/signup.feature [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+    When I submit a valid email
+    Then I am redirected to login success page
+
+    Given an unregistered user
+    When I submit an invalid email
+    Then I get warning message
+
+    Given an unregistered user
+    When I submit a valid twitter credentials
+    Then I am redirected to login welcome page
```
Comment (rimikt) [inline projects/courselets/features/signup.feature]: Add blank line

File: projects/courselets/features/steps/signup.py [hunk 14]
```diff
@@ -0,0 +1,70 @@
+from projects.courselets.features.utils.temp_mail import email_confirmation
+from behave import given, when, then
```
Comment (rimikt) [inline projects/courselets/features/steps/signup.py]: Please refactor imports according pep8

File: projects/courselets/features/steps/signup.py [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+@when('I submit a valid twitter credentials')
+def step_impl(context):
+    br = context.browser
+    br.get(context.base_url + 'login/')
+    LoginPage(context).twitter(USERNAME_TWITTER, PASSWORD_TWITTER)
+    time.sleep(3)
+
+
+@then('I am redirected to login welcome page')
```
Comment (rimikt) [inline projects/courselets/features/steps/signup.py]: Duplicate of `@then('I am redirected to login success page')`

File: projects/courselets/features/pages/login_page.py [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+"""
+from .base_page import BasePage
+import time
+
+
+class LoginPage(BasePage):
+
+    def signup_form(self, email):
+        email_button = self.click(attr_name='body > div.container > div.buttons > div.container-fluid > div > a')
```
Comment (rimikt) [inline projects/courselets/features/pages/login_page.py]: As a next step I want you to add `data-test` attrs to navigate easy.

File: .idea/tanabata.iml [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (DenisKolesnik) [reply .idea/tanabata.iml]: done

File: envs/local.env (hunk 1 above)
Comment (DenisKolesnik) [reply envs/local.env]: done

File: envs/local.env.example (hunk 2 above)
Comment (DenisKolesnik) [reply envs/local.env.example]: done

File: projects/courselets/features/pages/main_page.py (hunk 12 above)
Comment (DenisKolesnik) [reply projects/courselets/features/pages/main_page.py]: Done

File: projects/courselets/features/signup.feature (hunk 13 above)
Comment (DenisKolesnik) [reply projects/courselets/features/signup.feature]: Done

File: projects/courselets/features/steps/signup.py (hunk 15 above)
Comment (DenisKolesnik) [reply projects/courselets/features/steps/signup.py]: done
